addon/mixins/resize-text.js

```diff
@@ -23,24 +41,34 @@ export default Mixin.create({
   },
 
   scaleFont() {
-    const style = getComputedStyle(this.element);
-    const minSize = this.get('minSize');
-    const maxSize = this.get('maxSize');
-    const container = this.get('containerElement');
-    let elementWidth;
-    if (container) {
-      elementWidth = container.clientWidth;
-    } else {
-      elementWidth = this.element.clientWidth;
-    }
+    const container = this.get('containerElement') || this.element;
+    const style = getComputedStyle(container);
+    const currentFontSize = extractPixels(style.fontSize);
+
+    // if shrink or grow are set to false, use the currentFontSize
+    const { shrink, grow } = this.getProperties('shrink', 'grow');
+    const minSize = !shrink ? currentFontSize : this.get('minSize');
+    const maxSize = !grow ? currentFontSize : this.get('maxSize');
+
+    // the computed style.width is the given width, w/o padding, which is what we want
+    let elementWidth = extractPixels(style.width) || container.clientWidth;
+    // remove the padding, which is the actual inner/renderable width
+    elementWidth -= extractPixels(style.paddingLeft);
+    elementWidth -= extractPixels(style.paddingRight);
+
     let fontSize = this.get('textMeasurer')
-        .fitTextSize(this.element.innerText, elementWidth, style.fontStyle + ' 14px ' + style.fontFamily);
+        .fitTextSize(this.element.innerText, elementWidth,
+          `${style.fontStyle} ${style.fontSize} ${style.fontFamily}`);
     if (fontSize > maxSize) {
       fontSize = maxSize;
     } else if (fontSize < minSize) {
       fontSize = minSize;
     }
-    this.element.style['font-size'] = fontSize + 'px';
+
+    // only set the style if it actually changed
```

Comment:
Doesn't ember already handle this?

Reply:
I can double-check, but I believe that since you are modifying the actual `Element` properties, this falls outside of what Ember "manages". I'll debug and see what happens. :+1: